build.go

```diff
@@ -0,0 +1,355 @@
+		}
+
+		if composerGlobalBin != "" {
+			path = strings.Join([]string{
+				composerGlobalBin,
+				path,
+			}, string(os.PathListSeparator))
+		}
+
+		workspaceVendorDir := filepath.Join(context.WorkingDir, "vendor")
+
+		if value, found := os.LookupEnv(ComposerVendorDir); found {
+			workspaceVendorDir = filepath.Join(context.WorkingDir, value)
+		}
+
+		composerPackagesLayer, layerVendorDir, err := runComposerInstall(
+			logger,
+			context,
+			composerInstallOptions,
+			composerPhpIniPath,
+			path,
+			composerInstallExec,
+			workspaceVendorDir,
+			calculator)
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
+
+		logger.Process("Writing symlink to %s", workspaceVendorDir)
+
+		err = os.Symlink(layerVendorDir, workspaceVendorDir)
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
+
+		if os.Getenv(BpLogLevel) == "DEBUG" {
+			logger.Debug.Action("listing files in %s:", layerVendorDir)
+			files, err := ioutil.ReadDir(layerVendorDir)
+			if err != nil {
+				return packit.BuildResult{}, err
+			}
+			for _, f := range files {
+				logger.Debug.Detail(fmt.Sprintf("- %s", f.Name()))
+			}
+		}
+
+		err = runCheckPlatformReqs(logger, checkPlatformReqsExec, context.WorkingDir, composerPhpIniPath, composerPackagesLayer.Path, path)
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
+
+		return packit.BuildResult{
+			Layers: []packit.Layer{
+				composerPackagesLayer,
+			},
+		}, nil
+	}
+}
+
+func runComposerGlobalIfRequired(
+	logger scribe.Emitter,
+	context packit.BuildContext,
+	composerGlobalExec Executable,
+	path string,
+	composerPhpIniPath string) (string, error) {
+	composerInstallGlobal, found := os.LookupEnv(BpComposerInstallGlobal)
+
+	if !found {
+		return "", nil
+	}
+
+	logger.Process("Running 'composer global require'")
+
+	composerGlobalLayer, err := context.Layers.Get(ComposerGlobalLayerName)
+	if err != nil {
+		return "", err
+	}
+
+	composerGlobalLayer, err = composerGlobalLayer.Reset()
+	if err != nil {
+		return "", err
+	}
+
+	globalPackages := strings.Split(composerInstallGlobal, " ")
+
+	composerGlobalBuffer := bytes.NewBuffer(nil)
+
+	execution := pexec.Execution{
+		Args: append([]string{"global", "require", "--no-progress"}, globalPackages...),
+		Dir:  composerGlobalLayer.Path,
+		Env: append(os.Environ(),
+			"COMPOSER_NO_INTERACTION=1", // https://getcomposer.org/doc/03-cli.md#composer-no-interaction
+			fmt.Sprintf("COMPOSER_HOME=%s", composerGlobalLayer.Path),
+			fmt.Sprintf("PHPRC=%s", composerPhpIniPath),
+			"COMPOSER_VENDOR_DIR=vendor", // ensure default in the layer
+			fmt.Sprintf("PATH=%s", path),
+		),
+		Stdout: composerGlobalBuffer,
+		Stderr: composerGlobalBuffer,
+	}
+	err = composerGlobalExec.Execute(execution)
+
+	if err != nil {
+		logger.Subprocess(composerGlobalBuffer.String())
+		return "", err
+	}
+
+	composerGlobalBin := filepath.Join(composerGlobalLayer.Path, "vendor", "bin")
+
+	if os.Getenv(BpLogLevel) == "DEBUG" {
```

Comment:
This shouldn't be necessary, since the logger passed in would already have `DEBUG` set, and the `logger.Debug` logs will only show up in that case

Reply:
The purpose was to avoid reading from the filesystem if the logs would be ignored anyways. It's common to see in Java logging setups but a bit awkward here. I'll see if there's A Better Way ©.

Reply:
Alright, that's fine